static/js/components/MultipleClassificationsForm.jsx

```diff
@@ -0,0 +1,363 @@
+  const sortedClassifications = getSortedClasses(currentClassifications);
+
+  // For each existing taxonomy/classification, update initial sliders
+  sortedClassifications.forEach((classifications) => {
+    classifications.forEach((classification) => {
+      // Take just the latest values for each field
+      if (
+        !initialFormState[classification.taxonomy_id][
+          classification.classification
+        ]
+      ) {
+        initialFormState[classification.taxonomy_id][
+          classification.classification
+        ] = { depth: -1, probability: classification.probability };
+      }
+    });
+  });
+  const [formState, setFormState] = useState(initialFormState);
+
+  const getNode = (classification, path) => {
+    // Get node from hierarchy
+    let node;
+    let hierarchy = selectedTaxonomy?.hierarchy.subclasses;
+    const pathCopy = [...path];
+
+    while (pathCopy.length > 0) {
+      const ancestor = pathCopy.pop();
+      node = hierarchy?.find((x) => x.class === ancestor);
+      hierarchy = node?.subclasses;
+    }
+    node = hierarchy?.find((x) => x.class === classification);
+    return node;
+  };
+
+  const sumChildren = (node, newFormState) => {
+    // Sum the probabilities of the children
+    const children = node?.subclasses;
+    let sum = 0;
+    children?.forEach((subclass) => {
+      sum +=
+        newFormState[selectedTaxonomy.id][subclass.class]?.probability || 0;
+    });
+
+    return sum;
+  };
+
+  const updateChildren = (classification, newValue, newFormState, depth) => {
+    classification.subclasses?.forEach((subclass) => {
+      newFormState[selectedTaxonomy.id][subclass.class] = {
+        depth,
+        probability: Math.min(
+          newValue,
+          newFormState[selectedTaxonomy.id][subclass.class]?.probability || 0
+        ),
+      };
+      updateChildren(subclass, newValue, newFormState);
+    });
+  };
+
+  const handleChange = (newValue, classification, path) => {
+    const newFormState = { ...formState };
+    newFormState[selectedTaxonomy.id][classification] = {
+      depth: path.length,
+      probability: newValue,
+    };
+
+    // Update higher-level classification probabilities to be
+    // the maximum between the current probability and the sum
+    // of the subclasses' probabilities.
+    path.forEach((ancestor, i) => {
+      const probabilityOfSubclasses = Math.min(
+        sumChildren(getNode(ancestor, path.slice(i + 1)), newFormState),
+        1
+      );
+      newFormState[selectedTaxonomy.id][ancestor] = {
+        depth: path.slice(i + 1).length,
+        probability: Math.max(
+          probabilityOfSubclasses,
+          newFormState[selectedTaxonomy.id][ancestor]?.probability || 0
+        ),
+      };
+    });
+
+    // Update subclasses' probabilities to be the minimum
+    // between the current probability and the new value
+    const node = getNode(classification, path);
+    updateChildren(node, newValue, newFormState, path.length);
```

Comment:
Let me know if this behavior is what you/users would expect. I took some creative liberty here based on what I thought was logical, but not sure what the exact behavior is in the current marshal

Reply:
The logic makes sense to me. @mcoughlin you agreez?
Poked around with the ui, and seeing some weird behavior at time though:
![class-selectors](https://user-images.githubusercontent.com/7557205/125144860-1b430e00-e0d4-11eb-901b-b891db629257.gif)

Reply:
@kmshin1397 @dmitryduev First of all, playing around with it is awesome. I understand based on this that the parent classification defaults to the sum of the children one. And the child classification cannot exceed that of the parent. I like this behavior as it enforces conservation of classification ;) I am wondering at the children level, if I choose now like 0.75 and then 0.5 for another,  W Uma and detached eclipsing say, eclipsing jumps to 1.0 which makes sense. I am wondering if by default it should bounce back to 0.5+0.5. I think how it *should* behave is a bit ill-defined anyways and seems clear you can get to where it should be without much difficulty.

Reply:
Sure, that makes sense and should be doable. I'll work on that change.